packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -505,6 +509,16 @@ class _DraggableSheetExtent {
   // this because we don't want to snap away from an initial or programmatically set size.
   bool hasDragged;
 
+  // Used to determine if the sheet should move to the new initial size when it
+  // changes.
+  // We need both `hasChanged` and `hasDragged` to achieve the following
+  // behavior:
+  //   1. The sheet should only snap following user drags (as opposed to
+  //      programmatic sheet changes.
```

Comment:
Is this publicly documented somewhere?

Reply:
I think this behavior probably makes sense but we should document it.

Reply:
done